src/Extension/Core/Widget/ChartRenderer.php

```diff
@@ -65,19 +65,20 @@ public function render(WidgetRenderer $renderer, Widget $widget, Buffer $buffer)
                 ->setStyle($widget->yAxis->style);
         }
 
-        foreach ($widget->dataSets as $dataSet) {
-            $subBuffer = Buffer::empty($layout->graphArea);
-            $renderer->render($renderer, CanvasWidget::default()
+        $widgets = array_map(function (DataSet $dataSet) use ($widget): CanvasWidget {
+            return CanvasWidget::default()
                 ->backgroundColor($widget->style->bg ?? AnsiColor::Reset)
                 ->xBounds($this->resolveBounds($dataSet, $widget->xAxis->bounds, 0))
                 ->yBounds($this->resolveBounds($dataSet, $widget->yAxis->bounds, 1))
                 ->marker($dataSet->marker)
                 ->paint(function (CanvasContext $context) use ($dataSet): void {
                     $context->draw(PointsShape::new($dataSet->data, $dataSet->style->fg ?? AnsiColor::Reset));
-                }), $subBuffer);
-            $buffer->putBuffer($layout->graphArea->position, $subBuffer);
+                });
+        }, $widget->dataSets);
 
-        }
+        $subBuffer = Buffer::empty($layout->graphArea);
+        $renderer->render($renderer, CompositeWidget::fromWidgets(...$widgets), $subBuffer);
+        $buffer->putBuffer($layout->graphArea->position, $subBuffer);
```

Comment:
not sure this needs a "sub buffer"? the buffer passed to the widget should already be empty

Reply:
You mean simply do this?

```php
$renderer->render($renderer, CompositeWidget::fromWidgets(...$widgets), $buffer);
```

But this causes the chart to render incorrectly:

![image](https://github.com/php-tui/php-tui/assets/999232/65490e2e-ca80-4fb2-9fb4-8c2310ca3f18)

And I haven't studied this part enough to be able to fix it. Do you know what we should do here?

Reply:
i see , ignore me :)